drivers/wifi/nxp/nxp_wifi_drv.c

```diff
@@ -1177,9 +1181,33 @@ static int nxp_wifi_status(const struct device *dev, struct wifi_iface_status *s
 }
 
 #if defined(CONFIG_NET_STATISTICS_WIFI)
+#ifdef CONFIG_NXP_WIFI_GET_LOG
+static int nxp_wifi_get_detail_stats(int bss_type, wlan_pkt_stats_t *stats)
+{
+	int ret = -1;
+
+	if (bss_type == WLAN_BSS_TYPE_STA) {
+		ret = wlan_get_log(stats);
+	}
+#ifdef CONFIG_NXP_WIFI_SOFTAP_SUPPORT
+	else if (bss_type == WLAN_BSS_TYPE_UAP) {
+		ret = wlan_uap_get_log(stats);
+	}
+#endif
+	else {
+		ret = -ENODEV;
+	}
+	return ret;
+}
+#endif
+
 static int nxp_wifi_stats(const struct device *dev, struct net_stats_wifi *stats)
 {
 	struct interface *if_handle = (struct interface *)dev->data;
+#ifdef CONFIG_NXP_WIFI_GET_LOG
+	int ret = -1;
```

Comment:
No need to set the value to -1 as you assign value to it almost immediately. Same thing with `wifi_stats` which gets its value set also few lines below.